docs/src/dev/profiling.md

```diff
@@ -60,47 +60,80 @@ Fields and their meaning are as follows:
 
 ## Perf
 
-Note that `yk-config --cflags` includes `-Wl,--no-rosegment`, which [gives
-better profiling
-information](https://github.com/flamegraph-rs/flamegraph#cargo-flamegraph) for
-binaries linked with `lld` (as all yk C interpreters are). 
+On Linux, `perf` can be used to profile yk. You must first need to record an
+execution of an interpreter and then separately view the profiling data that
+was generated.
 
-One way to view profile data is with perf.
+
+### Recording a Profile
+
+To record a profile we first recommend compiling yk with debugging info
+embedded. cargo's `debug` profile does this automatically, but because no code
+optimisation is performed, the profiles are unrepresentative. We recommend
+using yk's provided `profile-with-debug` profile, which turns on
+`--release`-style code optimisation *and* embeds debugging information:
+
+```
+$ cargo build --profile=release-with-debug
+```
+
+Ensure that the interpreter you are profiling links to the appropriate version
+of yk and then call `perf record`:
 
 ```
 $ perf record --call-graph dwarf -g ./interpreter ...args...
-$ perf report -G --no-inline
```

Comment:
Did you mean to uninvert the call graph?

Did you mean to use bare symbols (and not DWARF symbols, which in my experience gives much better profiles)?

Reply:
Je ne parlez pas votres question.

Reply:
Second part of my comment was nonsense.

Reply:
Ah, `-G` was in the original doc and I copied it blindly. Fixed in f690220.